File: MapGUI3/src/Model/Statements/openRFile.java
package Model.Statements;

import Model.ProgramState;
import Model.FileTable;
import Model.SymbolTable;
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.AbstractMap;

public class openRFile implements IStatement {
    private String id;
    private String filename;


    public openRFile(String id, String filename) {
        this.id = id;
        this.filename = filename;
    }

    public ProgramState execute(ProgramState ps) {
        FileTable ft=ps.getFileTable();
        SymbolTable<String, Integer> symTable = ps.getSymTable();
        if(ft.exists(filename))
            return ps;
        try{
            BufferedReader buff = new BufferedReader(new FileReader(filename));
            int fd = ft.put(new AbstractMap.SimpleEntry<>(filename, buff));
            symTable.put(id, fd);
        } catch (IOException e) {
            e.printStackTrace();
        }
        return ps;
    }

    public String toString(){
        return "openRFile(" +id+','+filename+")";
    }
}

File: MapA7/MapA7/Model/Statements/OpenRFile.cs
using MapA7.Model.Exceptions;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MapA7.Model.Statements
{
    public class OpenRFile:IStatement
    {
        private String var;
        private String filename;

        public OpenRFile(String var, String filename)
        {
            this.var = var;
            this.filename = filename;
        }

        public ProgramState Execute(ProgramState ps)
        {
            SymbolTable symTable = ps.GetSymbolTable();
            FileTable fileTable = ps.GetFileTalbe();
            if (fileTable.Exists(filename))
                throw new CustomException("File " + filename + " is already opened.");
            try
            {
                StreamReader sr = new StreamReader(new FileStream(filename, FileMode.Open, FileAccess.Read));
                int fd = fileTable.Add(new System.Tuple<string, StreamReader>(filename, sr));
                symTable.Put(var, fd);
            }
            catch (IOException)
            {
                throw new CustomException("Opening of the file " + filename + " resultet int an error.");
            }
            return ps;
        }

        public String Tostring()
        {
            return "OpenRFile(" + var + ", " + filename + ")";
        }
    }
}
